src/ansys/pytwin/evaluate/twin_model.py

```diff
@@ -809,6 +1079,94 @@ def get_rom_directory(self, rom_name):
+            If the provided ROM name is not available.
+
+        Examples
+        --------
+        >>> from pytwin import TwinModel
+        >>> model = TwinModel(model_filepath='path_to_twin_model_with_TBROM_in_it.twin')
+        >>> model.initialize_evaluation()
+        >>> model.get_rom_nslist(model.tbrom_names[0])
+        """
+        self._log_key = "GetRomNamedSelectionList"
+
+        if not self.evaluation_is_initialized:
+            msg = "Twin model has not been initialized. "
+            msg += "Initialize evaluation before calling this method."
+            self._raise_error(msg)
+
+        if self.tbrom_info is None:
+            self._raise_error("Twin model does not include any TBROMs.")
+
+        if rom_name not in self.tbrom_names:
+            msg = f"The provided ROM name {rom_name} has not been found in the available TBROM names. "
+            msg += f"Call this method with a valid TBROM name."
+            msg += f"\n Available TBROM names are: {self.tbrom_names}."
+            self._raise_error(msg)
+
+        tbrom = self._tbroms[rom_name]
+
+        return tbrom.nsnames
+
+    def get_rom_inputfieldsnames(self, rom_name):
+        """
+        Get the list of input fields names associated to the TBROM named rom_name
+
+        Parameters
+        ----------
+        rom_name : str
+            Name of the ROM. To get a list of available ROMs, see the
+            :attr:`pytwin.TwinModel.tbrom_names` attribute.
+
+        Raises
+        ------
+        TwinModelError:
+            If ``TwinModel`` object has not been initialized.
+            If ``TwinModel`` object does not include any TBROMs.
+            If the provided ROM name is not available.
+
+        Examples
+        --------
+        >>> from pytwin import TwinModel
+        >>> model = TwinModel(model_filepath='path_to_twin_model_with_TBROM_in_it.twin')
+        >>> model.initialize_evaluation()
+        >>> model.get_rom_nslist(model.tbrom_names[0])
+        """
+        self._log_key = "GetRomDirectory"
+
+        if not self.evaluation_is_initialized:
+            msg = "Twin model has not been initialized. "
+            msg += "Initialize evaluation before calling this method."
+            self._raise_error(msg)
+
+        if self.tbrom_info is None:
+            self._raise_error("Twin model does not include any TBROMs.")
+
+        if rom_name not in self.tbrom_names:
+            msg = f"The provided ROM name {rom_name} has not been found in the available TBROM names. "
+            msg += f"Call this method with a valid TBROM name."
+            msg += f"\n Available TBROM names are: {self.tbrom_names}."
+            self._raise_error(msg)
+
+        tbrom = self._tbroms[rom_name]
+
+        return tbrom.nameinputfields
+
     def get_snapshot_filepath(self, rom_name: str, evaluation_time: float = 0.0):
         """
         Get the snapshot file that was created by the given ROM at the given time instant.
```

Comment:
```suggestion
        Get the snapshot file that was created by a ROM at a given time instant.
```
We try to use given as much as possible. The fact that you provide the ROM name implies that the ROM is given.